az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -366,41 +362,22 @@ public static void proxyUserKillAllSpawnedHadoopJobs(final String logFilePath,
         proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
           @Override
           public Void run() throws Exception {
-            HadoopJobUtils.killAllSpawnedHadoopJobs(logFilePath, log, jobProps);
+            Set<String> allSpawnedJobAppIDs = findApplicationIdFromLog(logFilePath, log);
+            YarnClient yarnClient = YarnUtils.createYarnClient(jobProps);
+            YarnUtils.killAllAppsOnCluster(yarnClient, allSpawnedJobAppIDs, log);
             return null;
           }
         });
       } else {
-        HadoopJobUtils.killAllSpawnedHadoopJobs(logFilePath, log, jobProps);
```

Comment:
you have repeat the code in the if and else branch, maybe you can extract it to a method

Reply:
Also why do you remove killAllSpawnedHadoopJobs? these repeated code should be in this method, right?

Reply:
oh it looks like 3 lines for now, if we want to add more logic into it in the future, it will make sense to extract it to a method, like killAllSpawnedHadoopJobs. It is good practice.

Reply:
I agree, will do in next PR

Reply:
I will prefer if we do it right in this PR only. Its a small change.